src/nvim/eval.c

```diff
@@ -514,6 +514,10 @@ void eval_init(void)
 
   for (size_t i = 0; i < ARRAY_SIZE(vimvars); i++) {
     p = &vimvars[i];
+    if (STRLEN(p->vv_name) > 16) {
+      EMSG("INTERNAL: name too long, increase size of dictitem16_T");
+      getout(1);
```

Comment:
@jamessan Agreed, but how about `abort()` so that Release builds are protected as well. `assert()` should be for things that "cannot happen".